File: 백준/cpp/boj_10845.cpp
#include <iostream>

using namespace std;

int queue[10000];
int f = 0, r = 0;

void push(int num)
{
  queue[r++] = num;
}

int empty()
{
  if (r - f == 0)
    return 1;
  else
    return 0;
}

int pop()
{
  if (!empty())
    return queue[f++];
  else
    return -1;
}

int size()
{
  return r - f;
}

int front()
{
  if (!empty())
    return queue[f];
  else
    return -1;
}

int back()
{
  if (!empty())
    return queue[r - 1];
  else
    return -1;
}

int main()
{
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  int n;
  cin >> n;
  for (int i = 0; i < n; i++)
  {
    string cmd;
    cin >> cmd;
    if (cmd == "push")
    {
      int num;
      cin >> num;
      push(num);
    }
    else if (cmd == "pop")
      cout << pop() << '\n';
    else if (cmd == "size")
      cout << size() << '\n';
    else if (cmd == "empty")
      cout << empty() << '\n';
    else if (cmd == "front")
      cout << front() << '\n';
    else if (cmd == "back")
      cout << back() << '\n';
  }

  return 0;
}

File: 백준/js/boj_10845.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const queue = [];
const answer = [];
const len = Number(input[0]);

for(let i=1; i<=len; i++){
    let cmd = input[i].split(' ');
    switch(cmd[0]) {
        case 'push':
            queue.push(cmd[1]);
            break;
        case 'pop':
            answer.push(queue.shift() || -1);
            break;
        case 'size':
            answer.push(queue.length);
            break;
        case 'empty':
            answer.push(queue[0] ? 0 : 1);
            break;
        case 'front':
            answer.push(queue[0] || -1);
            break;
        case 'back':
            answer.push(queue[queue.length-1] || -1);
            break;
    }
}

console.log(answer.join('\n'));